### PR TITLE
Onboarding LogicApp, APIM, Certificates and Domain Resources to Generic Platform

### PR DESCRIPTION
**Pull request by ShekharGupta1988**

### CODE REVIEW

**Comment by ShekharGupta1988** [comment]
Fixing tests and adding few more for new Resources

**Comment by ShekharGupta1988** [comment]
The kusto Change is actually a breaking change.  So Tests are rightfully failing:
Existing scripts will fail with error :
(5,64): error CS0121: The call is ambiguous between the following methods or properties: 'IKustoDataProvider.ExecuteQuery(string, string, string)' and 'IKustoDataProvider.ExecuteQuery(string, string, string, string)'

File: src/Diagnostics.ModelsAndUtils/Attributes/ResourceFilter/AppServiceCertificateFilter.cs [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+using System.Text;
+
+namespace Diagnostics.ModelsAndUtils.Attributes
+{
+    /// <summary>
+    /// Class representing Filter for App service certificate
+    /// </summary>
+    public class AppServiceCertificateFilter : ResourceFilterBase
+    {
+        public AppServiceCertificateFilter(bool internalOnly = true) : base(ResourceType.AppServiceCertificate, internalOnly)
```
Comment (stevenernst131) [inline src/Diagnostics.ModelsAndUtils/Attributes/ResourceFilter/AppServiceCertificateFilter.cs]: Are these constructors necessary?

Comment (ShekharGupta1988) [reply src/Diagnostics.ModelsAndUtils/Attributes/ResourceFilter/AppServiceCertificateFilter.cs]: Ya, it is. The base constructor has the Resource Type

Comment (stevenernst131) [reply src/Diagnostics.ModelsAndUtils/Attributes/ResourceFilter/AppServiceCertificateFilter.cs]: True, not sure you need internalOnly as a parameter

**Pull request merged**